src/execution_plan/ops/op_create.c

```diff
@@ -87,40 +89,40 @@ OpBase* NewCreateOp(RedisModuleCtx *ctx, GraphContext *gc, AST_Query *ast, Query
     return (OpBase*)op_create;
 }
 
-void _CreateNodes(OpCreate *op, Record *r) {
+void _CreateNodes(OpCreate *op, Record r) {
     Graph *g = op->gc->g;
     for(int i = 0; i < op->node_count; i++) {
         /* Get specified node to create. */
-        Node *n = op->nodes_to_create[i].original_node;
+        Node *n = op->nodes_to_create[i].node;
 
         /* Create a new node. */
         Node *newNode = Node_New(n->label, n->alias);
 
         /* Save node for later insertion. */
-        Vector_Push(op->created_nodes, newNode);
+        op->created_nodes = array_append(op->created_nodes, newNode);
 
         /* Update record with new node. */
-        Record_AddEntry(r, newNode->alias, SI_PtrVal(newNode));
+        Record_AddScalar(r, op->nodes_to_create[i].node_rec_idx, SI_PtrVal(newNode));
```

Comment:
Why aren't we using Record_AddNode and `*newNode`?

Reply:
It might be that we can, I'll have to double check.

Reply:
Reason why we use a pointer to the newly created node, is because of the way edges are created,
newly created edges will retrieve their src and dest nodes from a record, they must point to the original nodes.